### PR TITLE
Upgraded vscode-extension-vadimcn-vscode-lldb to the latest version

### PR DESCRIPTION
**Pull request by nigelgbanks**
###### Motivation for this change

Upgrading the package to its latest release.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
  - [x] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by nigelgbanks** [comment]
The maintainer of https://github.com/vadimcn/vscode-lldb/releases needs to correct the commits on the tagged release for this to be properly fixed. I will update this pull afterwards.

**Comment by nigelgbanks** [comment]
The issue https://github.com/vadimcn/vscode-lldb/issues/637 was resolved and I've updated the pull request accordingly.

**Comment by oxalica** [comment]
The commit message is outdated. Please amend it. Other diff LGTM.
Also, I'm not using vscode anymore. Could you please replace me in `meta.maintainers`?

---

Result of `nixpkgs-review pr 158299` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vscode-extensions.vadimcn.vscode-lldb</li>
  </ul>
</details>

**Comment by nigelgbanks** [comment]
> The commit message is outdated. Please amend it. Other diff LGTM. Also, I'm not using vscode anymore. Could you please replace me in `meta.maintainers`?

Sure thing I've altered the commit message and removed you as a maintainer adding myself.

File: pkgs/misc/vscode-extensions/vscode-lldb/default.nix [hunk 1]
```diff
@@ -98,7 +98,7 @@ in stdenv.mkDerivation {
     description = "A native debugger extension for VSCode based on LLDB";
     homepage = "https://github.com/vadimcn/vscode-lldb";
     license = with licenses; [ mit ];
-    maintainers = with maintainers; [ oxalica ];
+    maintainers = with maintainers; [ nigelgbanks ];
```
[the remark this answers is not in the document]
Comment (nigelgbanks) [reply pkgs/misc/vscode-extensions/vscode-lldb/default.nix:101]: My bad, I should have repeated all the steps in the template on every push. Thanks!

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 158299 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>1 package built:</summary>  <ul>    <li>vscode-extensions.vadimcn.vscode-lldb</li>  </ul></details>

**Pull request merged**